frontend/src/pages/modelServing/screens/global/InferenceServiceServingRuntime.tsx

```diff
@@ -4,42 +4,73 @@ import { ServingRuntimeKind } from '#~/k8sTypes';
 import {
   getDisplayNameFromServingRuntimeTemplate,
   getServingRuntimeVersion,
+  getTemplateNameFromServingRuntime,
 } from '#~/pages/modelServing/customServingRuntimes/utils';
-import { SERVING_RUNTIME_SCOPE } from '#~/pages/modelServing/screens/const';
+import {
+  SERVING_RUNTIME_SCOPE,
+  ServingRuntimeVersionStatusLabel,
+} from '#~/pages/modelServing/screens/const';
 import ServingRuntimeVersionLabel from '#~/pages/modelServing/screens/ServingRuntimeVersionLabel';
 import ScopedLabel from '#~/components/ScopedLabel';
+import { useTemplateByName } from '#~/pages/modelServing/customServingRuntimes/useTemplateByName';
+import ServingRuntimeVersionStatus from '#~/pages/modelServing/screens/ServingRuntimeVersionStatus';
+import { getServingRuntimeVersionStatus } from '#~/pages/modelServing/utils';
 
 type Props = {
   servingRuntime?: ServingRuntimeKind;
   isProjectScoped?: boolean;
 };
 
-const InferenceServiceServingRuntime: React.FC<Props> = ({ servingRuntime, isProjectScoped }) => (
-  <>
-    {servingRuntime ? (
-      <Stack>
-        <StackItem>{getDisplayNameFromServingRuntimeTemplate(servingRuntime)}</StackItem>
-        <StackItem>
-          <LabelGroup>
-            {getServingRuntimeVersion(servingRuntime) && (
-              <ServingRuntimeVersionLabel
-                version={getServingRuntimeVersion(servingRuntime)}
-                isCompact
-              />
-            )}
-            {isProjectScoped &&
-              servingRuntime.metadata.annotations?.['opendatahub.io/serving-runtime-scope'] ===
-                SERVING_RUNTIME_SCOPE.Project && (
-                <ScopedLabel isProject color="blue" isCompact>
-                  Project-scoped
-                </ScopedLabel>
+const InferenceServiceServingRuntime: React.FC<Props> = ({ servingRuntime, isProjectScoped }) => {
+  const [versionStatus, setVersionStatus] = React.useState<string | undefined>(undefined);
+  const templateName = servingRuntime
+    ? getTemplateNameFromServingRuntime(servingRuntime)
+    : undefined;
+
+  const [template, templateLoaded, templateError] = useTemplateByName(templateName);
+
+  React.useMemo(() => {
+    if (templateLoaded && !templateError && servingRuntime) {
+      const servingRuntimeVersion = getServingRuntimeVersion(servingRuntime);
+      const templateVersion = getServingRuntimeVersion(template);
+      setVersionStatus(getServingRuntimeVersionStatus(servingRuntimeVersion, templateVersion));
+    }
+  }, [template, templateLoaded, templateError, servingRuntime]);
```

Comment:
you should just be able to do this:
```suggestion
  const version = React.useMemo(() => {
    if (templateLoaded && !templateError && servingRuntime) {
      const servingRuntimeVersion = getServingRuntimeVersion(servingRuntime);
      const templateVersion = getServingRuntimeVersion(template);
      setVersionStatus(getServingRuntimeVersionStatus(servingRuntimeVersion, templateVersion));
    }
  }, [template, templateLoaded, templateError, servingRuntime]);
```
and delete L25

Reply:
And change line 36 to `return getServingRuntimeVersionStatus(servingRuntimeVersion, templateVersion)`

`React.useMemo` is for creating a memoized variable. It should be used very similarly to normal variable assignment `const version = getVersion(...)` and creation. But `useMemo` will only rerun / calculate when the dependency array is updated. A non-memoized variable would recalculate on everysingle rerender.